containers/podman-desktop
# Search images in registry in Image Pull form

Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Search images in registry in Image Pull form

Two fields:
- first field (required) to enter a search term, the list of image names found will be listed. If the search fails, the text entered is displayed in the list and selectable, so the user can pull an image not discoverable by typing its full name (including its tag)
- second field (optional) lists the tags found for the image selected in the first field 

### Screenshot / video of UI

- search and pull image from docker.io with default tag
- search and pull image from docker.io with specific tag
- search and pull image from quay.io
- pull non discoverable image from ghcr.io + discovered tag
- pull image+tag from ghcr.io 

https://github.com/user-attachments/assets/e3b4ce4d-8ea7-4bfd-9e3c-06a8ef040469



### What issues does this PR fix or reference?

Fixes #971 

### How to test this PR?

Go to the Images list page, press Pull button, then search images (see video)


e2e tests have been modified to support the new system. A new parameter has been added to `pullImage`, two parameters are now necessary:
- `searchTerm`: the text entered in the search input
- `imageName`: the item selected in the result list
